modules/gateway/helpers.go

```diff
@@ -1,57 +1,60 @@
 package gateway
 
 import (
-	"errors"
 	"io/ioutil"
-	"math/rand"
-	"sync"
+	"net"
+	"net/http"
 
 	"github.com/NebulousLabs/Sia/encoding"
-	"github.com/NebulousLabs/Sia/network"
+	"github.com/NebulousLabs/Sia/modules"
 )
 
-func (g *Gateway) addPeer(peer network.Address) error {
-	if _, exists := g.peers[peer]; exists {
-		return errors.New("peer already added")
-	}
-	g.peers[peer] = struct{}{}
-	return nil
+var pong = [4]byte{'p', 'o', 'n', 'g'}
+
+// Ping returns whether an Address is reachable and responds correctly to the
+// ping request -- in other words, whether it is a potential peer.
```

Comment:
Are there protections against other programs that might return 'pong'? If you're just calling "Ping" on a random ip address and port, it's not terribly far fetched that another program might return "pong". Perhaps we should add magic bytes here. ("siapong"?).